apps/next/src/pages/AccountManagement/components/Wallets/components/BulkDeleteButtons.tsx

```diff
@@ -44,7 +51,20 @@ export const BulkDeleteButtons: FC = () => {
             size="extension"
             fullWidth
             disabled={selectedAccounts.length === 0}
-            onClick={() =>
+            onClick={() => {
+              const selectedAccountsData = selectedAccounts
+                .map((id) => allAccounts.find((acc) => acc.id === id))
+                .filter((account): account is Account & { walletId?: string } =>
```

Comment:
`.filter(isPrimaryAccount)` should do the trick, available from `@core/common`:

https://github.com/ava-labs/core-extension/blob/a785d7022c8d63d2e9e74bdf7d49a6fde30925dc/packages/common/src/utils/accounts/accountTypeGuards.ts#L19-L21